### PR TITLE
Clash Map for predicting active conformations.

### PR DESCRIPTION
**Pull request by ssepeq**
There's a lot in this PR, but the main things are:

- Clash map generation that can be used to estimate the binding pocket changes for OR activation;
- Improved Best-Binding algorithm that groups ligand atoms and protein residues into "gloms", and aligns them;
- Infrastructure for active-first predictions;
- Misc fixes.

### CODE REVIEW

**Comment by ssepeq** [comment]
ADORA2_cmp

**Comment by ssepeq** [comment]
🟨🟩🟩

**Comment by primaryodors** [comment]
While the clash map is a really awesome development, and there is a lot in this PR worth saving, it would probably be best to review this as just docking quality improvements and move on to #195.

File: src/primarydock.cpp [hunk 1]
```diff
@@ -1451,19 +1796,176 @@ int main(int argc, char** argv)
...
+
+                    default:
+                    ;
+                }
+
+                bool taken = false;
+                for (j=0; j<3; j++) if (glomtmp.atoms == ligand_gloms[j].atoms) taken = true;
+                if (taken) continue;
+
+                bool toosame = false;
```
Comment (primaryodors) [inline src/primarydock.cpp:1928]: Should rename this to `too_similar` everywhere it occurs.

**Review by primaryodors — CHANGES_REQUESTED** [review]

**Review by primaryodors — APPROVED** [review]

**Comment by primaryodors** [comment]
🟩🟩🟩

**Pull request merged**